statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,681 @@
+
+        Returns
+        -------
+        hess : ndarray, (k_vars, k_vars)
+            The Hessian, second derivative of loglikelihood function,
+            evaluated at `params`
+
+        Notes
+        -----
+        """
+        hess_arr_main = self._hessian_main(params)
+        hess_arr_infl = self._hessian_inflate(params)
+
+        if hess_arr_main is None or hess_arr_infl is None:
+            return approx_hess(params, self.loglike)
+
+        dim = self.k_exog + self.k_inflate
+
+        hess_arr = np.zeros((dim, dim))
+
+        hess_arr[:self.k_inflate,:] = hess_arr_infl
+        hess_arr[self.k_inflate:,self.k_inflate:] = hess_arr_main
+
+        tri_idx = np.triu_indices(self.k_exog + self.k_inflate, k=1)
+        hess_arr[tri_idx] = hess_arr.T[tri_idx]
+
+        return hess_arr
+
+    def predict(self, params, exog=None, exog_infl=None, exposure=None,
+                offset=None, which='mean'):
+        """
+        Predict response variable of a count model given exogenous variables.
+
+        Notes
+        -----
+        If exposure is specified, then it will be logged by the method.
+        The user does not need to log it first.
+        """
+        if exog is None:
+            exog = self.exog
+            offset = getattr(self, 'offset', 0)
+            exposure = getattr(self, 'exposure', 0)
+
+        if exog_infl is None:
+            exog_infl = self.exog_infl
+
+        if exposure is None:
+            exposure = 0
+        elif exposure != 0:
+            exposure = np.log(exposure)
+
+        if offset is None:
+            offset = 0
+
+        params_infl = params[:self.k_inflate]
+        params_main = params[self.k_inflate:]
+
+        if self.infl == 'logit':
+            prob_poisson = 1 / (1 + np.exp(np.dot(exog_infl, params_infl)))
+        elif self.infl == 'probit':
+            raise NotImplemented('Predict for Probit inflation not implemented')
+
+        lin_pred = np.dot(exog, params_main[:self.exog.shape[1]]) + exposure + offset
+        
+        tmp_exog = self.model_main.exog
+        tmp_endog = self.model_main.endog
+        self.model_main.exog = exog
+        self.model_main.endog = np.zeros((exog.shape[0]))
+        llf = self.model_main.loglikeobs(params_main)
+        self.model_main.exog = tmp_exog
+        self.model_main.endog = tmp_endog
+
+        prob_zero = (1 - prob_poisson) + prob_poisson * np.exp(llf)
+
+        if which == 'mean':
+            return prob_poisson * np.exp(lin_pred)
+        elif which == 'poisson-mean':
+            return np.exp(lin_pred)
+        elif which == 'linear':
+            return lin_pred
+        elif which == 'mean-nonzero':
+            return prob_poisson * np.exp(lin_pred) / (1 - prob_zero)
+        elif which == 'prob-zero':
+            return prob_zero
+        elif which == 'prob':
+            return self._predict_prob(params)
+        else:
+            raise ValueError('keyword `which` not recognized')
```

Comment:
better: `raise ValueError('value of keyword `which` not recognized')`
I got confused and thought I misspelled "which" when I had used the wrong value (zero-prob instead of prob-zero)

Reply:
Will it be better like this? `raise ValueError('which = %s not handled' % which)`

Reply:
Yes sounds good, but I would use "available" instead of "handled", and add an "is"
"is not available"